### PR TITLE
74435 - Timed sync

### PR DESCRIPTION
**Pull request by hbjorgo**
Added a scheduled background service that will invoke synchronization commands.

### CODE REVIEW

**Comment by janid1967** [comment]
Please add PBI no in front of title of PR

File: src/Equinor.Procosys.Preservation.WebApi/Equinor.Procosys.Preservation.WebApi.csproj [hunk 1]
```diff
@@ -24,6 +24,7 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
     <PackageReference Include="Microsoft.Extensions.Configuration.AzureKeyVault" Version="3.1.4" />
+    <PackageReference Include="Microsoft.Identity.Client" Version="4.14.0" />
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.WebApi/Equinor.Procosys.Preservation.WebApi.csproj:27]: I guess this is intentionally

File: src/Equinor.Procosys.Preservation.WebApi/Misc/PlantProvider.cs [hunk 2]
```diff
@@ -1,33 +1,15 @@
 ﻿using System;
 using Equinor.Procosys.Preservation.Domain;
-using Microsoft.AspNetCore.Http;
 
 namespace Equinor.Procosys.Preservation.WebApi.Misc
 {
-    public class PlantProvider : IPlantProvider
+    public class PlantProvider : IPlantProvider, IPlantSetter
     {
         public const string PlantHeader = "x-plant";
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.WebApi/Misc/PlantProvider.cs:8]: Maybe rename PlantHeader to something to associate better to Http? Or move it to better place?

File: src/Equinor.Procosys.Preservation.WebApi/Synchronization/SynchronizationService.cs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        private readonly IMediator _mediator;
+        private readonly IClaimsProvider _claimsProvider;
+        private readonly IPlantSetter _plantSetter;
+        private readonly ICurrentUserSetter _currentUserSetter;
+        private readonly IBearerTokenSetter _bearerTokenSetter;
+        private readonly IClaimsTransformation _claimsTransformation;
+        private readonly IAuthenticator _authenticator;
+        private readonly IPlantCache _plantCache;
+
+        public SynchronizationService(
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.WebApi/Synchronization/SynchronizationService.cs:30]: Add a TECH Todo for writing unit tests later

File: src/tests/Equinor.Procosys.Preservation.Command.Tests/MiscCommands/Clone/PlantProvider.cs [hunk 4]
```diff
@@ -1,28 +1,23 @@
...
+using Equinor.Procosys.Preservation.Domain;
 
 namespace Equinor.Procosys.Preservation.Command.Tests.MiscCommands.Clone
 {
-    public class PlantProvider : IPlantProvider
+    public class PlantProvider : IPlantProvider, IPlantSetter
     {
-        private string _temporaryPlant;
-        private readonly string _plant;
+        public const string PlantHeader = "x-plant";
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Command.Tests/MiscCommands/Clone/PlantProvider.cs:8]: remove

File: src/tests/Equinor.Procosys.Preservation.Command.Tests/MiscCommands/Clone/PlantProvider.cs [hunk 5]
```diff
@@ -1,28 +1,23 @@
...
         {
-            get
-            {
-                if (_temporaryPlant != null)
-                {
-                    return _temporaryPlant;
-                }
-                return _plant;
-            }
+            _plant = plant;
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Command.Tests/MiscCommands/Clone/PlantProvider.cs:13]: expr.body

File: src/tests/Equinor.Procosys.Preservation.Command.Tests/MiscCommands/Clone/PlantProvider.cs [hunk 6]
```diff
@@ -1,28 +1,23 @@
...
         }
 
-        public void SetTemporaryPlant(string plant) => _temporaryPlant = plant;
+        public string Plant => _plant ?? throw new Exception("Could not determine current plant");
 
-        public void ReleaseTemporaryPlant() => _temporaryPlant = null;
+        public void SetPlant(string plant)
+        {
+            _plant = plant;
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Command.Tests/MiscCommands/Clone/PlantProvider.cs:20]: expr.body

File: src/Equinor.Procosys.Preservation.WebApi/Equinor.Procosys.Preservation.WebApi.csproj (hunk 1 above)
Comment (hbjorgo) [reply src/Equinor.Procosys.Preservation.WebApi/Equinor.Procosys.Preservation.WebApi.csproj:27]: Yes. It's needed to get a bearer token.

File: src/Equinor.Procosys.Preservation.WebApi/Misc/PlantProvider.cs (hunk 2 above)
Comment (hbjorgo) [reply src/Equinor.Procosys.Preservation.WebApi/Misc/PlantProvider.cs]: Moved. Ok?

File: src/Equinor.Procosys.Preservation.WebApi/Synchronization/SynchronizationService.cs (hunk 3 above)
Comment (hbjorgo) [reply src/Equinor.Procosys.Preservation.WebApi/Synchronization/SynchronizationService.cs:30]: Done

File: src/Equinor.Procosys.Preservation.WebApi/Misc/PlantProvider.cs (hunk 2 above)
Comment (janid1967) [reply src/Equinor.Procosys.Preservation.WebApi/Misc/PlantProvider.cs]: ok

**Pull request merged**